### PR TITLE
test(db): audit and harden reasoning effort migration

### PR DESCRIPTION
**Pull request by luodaoyi**
## Summary

- replace migration v18's full-table `COUNT(*)` with a bounded primary-key threshold probe before a potentially blocking MySQL fallback DDL
- make duplicate-column handling consistent across MySQL and non-MySQL paths
- validate reasoning metadata types and invalid Claude thinking budgets
- add complete migration, repository compatibility, historical NULL, idempotence, and metadata edge-case tests

## Database audit coverage

- SQLite actual migration and rerun
- historical rows remain SQL NULL and read as an empty API value
- repository Create, Update, GetByID, List, ListCursor, and ListActive with and without the new column
- MySQL 8 instant add-column success and duplicate-column race
- MySQL 5.7 guarded fallback for below/equal threshold, above threshold, probe failure, DDL failure, and duplicate-column race
- all statements in the v18 reasoning migration helpers and request metadata parser are covered

## Verification

- `go test ./internal/requestmeta ./internal/repository/sqlite ./internal/executor ./internal/handler`
- `go vet ./internal/requestmeta ./internal/repository/sqlite ./internal/executor ./internal/handler`
- targeted database and metadata tests repeated 20 times
- `go test ./...` passes all relevant packages, e2e, and multiinstance; on Windows only the pre-existing `cmd/maxx-cli/internal/cfg` permission and legacy-path tests fail, which are outside this PR

## Summary by CodeRabbit

* **Bug Fixes**
  * 改进推理强度信息解析，增强大小写、空白、类型异常及边界值处理。
  * 修复无效或缺失的推理配置可能产生错误默认值的问题。
  * 优化数据库升级流程，降低大数据量场景下的升级风险，并支持安全重试。
  * 确保历史请求记录及不同查询方式下的推理强度显示保持一致。
* **Tests**
  * 扩展多种数据库、模型格式及异常输入场景的验证覆盖。

### CODE REVIEW

File: internal/repository/sqlite/migrations.go [hunk 1]
```diff
@@ -576,36 +575,63 @@ func runReasoningEffortColumnMigration(db *gorm.DB) error {
...
 	}
 
 	if err := db.Exec(ddl).Error; err != nil {
-		lower := strings.ToLower(err.Error())
-		if strings.Contains(lower, "duplicate column") || strings.Contains(lower, "already exists") {
+		if isDuplicateColumnError(err) {
 			return nil
 		}
 		return err
 	}
```
[the remark this answers is not in the document]
Comment (luodaoyi) [reply internal/repository/sqlite/migrations.go:602]: 已在 d61dabb 修复：MySQL 受保护的回退 DDL 现在通过 GORM `Connection` 固定到同一物理连接，读取并临时设置 10 秒 `SESSION lock_wait_timeout`，DDL 成功或失败后都恢复原值。DDL 的 1205 超时错误会在恢复成功后原样返回；读取、设置、DDL 超时、恢复失败以及 DDL 失败后恢复失败均有测试，相关 v18 helper 语句覆盖率为 100%。

**Review by ymkiux — APPROVED** [review]
